### PR TITLE
A few more exception rules

### PR DESCRIPTION
**Pull request by schalkms**
This are the last few exception rules which I like to see in **detekt**.

### CODE REVIEW

**Comment by schalkms** [comment]
seems related to #333

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	override fun visitClassInitializer(initializer: KtClassInitializer) {
+		if (hasThrowExpression(initializer.body)) {
+			report(CodeSmell(issue, Entity.from(initializer)))
+		}
+	}
+
+	private fun isPotentialMethod(function: KtNamedFunction): Boolean {
+		return when (function.name) {
+			"toString", "hashCode", "equals", "finalize" -> true
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt]: Should we make them configurable?

**Comment by arturbosch** [comment]
Cool rules thanks for the big work on exceptions :)

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt (hunk 1 above)
Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt]: I don't think so. They are not expected to raise exceptions.

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt]: I mean that users can specify themself functions which should not throw exceptions?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ExceptionRaisedInUnexpectedLocation.kt]: I misunderstood your comment. Yes, definitely!

**Pull request merged**